src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourcesOptionsControl.cs

```diff
@@ -607,7 +609,15 @@ private void UpdateTextBoxes(PackageSourceContextInfo packageSource)
             }
         }
 
-        private void OnBrowseButtonClicked(object sender, EventArgs e)
+        private void OnBrowseButtonClicked(object sender, EventArgs args)
+        {
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async delegate
```

Comment:
You don't need an extra delegate here. 
See above comment.

Reply:
Ok. Addressed.